tests/containers/buildah_image.pm

```diff
@@ -0,0 +1,48 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Package: buildah
+# Summary: building OCI-compatible sle base images with buildah.
+# - install buildah
+# - create and run container
+# - build image with service and test connectivity
+# - cleanup system (images, containers)
+# Maintainer: qa-c team <qa-c@suse.de>
+
+use Mojo::Base qw(consoletest);
+use testapi;
+use utils;
+use containers::common;
+use containers::container_images;
+use suse_container_urls 'get_suse_container_urls';
+use version_utils qw(get_os_release check_os_release);
+
+sub run {
+    my ($image_names, $stable_names) = get_suse_container_urls();
+    my ($running_version, $sp, $host_distri) = get_os_release;
+    my $runtime = "buildah";
+
```

Comment:
Just add:
```perl
install_podman_when_needed($host_distri);
```

Reply:
if i get it right you suggest to keep each subroutine atomic. And then call both as needed. anyway i dont think it is urgent and i am intent to leave it as is for now. feel free to submit a PR for it.